packages/ui/src/common/utils/crypto/worker/client.ts

```diff
@@ -0,0 +1,33 @@
+import { uniqueId } from 'lodash'
+import { filter, firstValueFrom, fromEvent, Observable } from 'rxjs'
+
+import { compute, WorkerRequest, WorkerRequestType, WorkerResponse } from './utils'
+
+let worker: Worker
+let messages: Observable<MessageEvent<WorkerResponse>>
+
+export const hashFile = computeInWorker('HASH_FILE')
+
+export const merkleRootFromBinary = computeInWorker('MERKLE_ROOT')
+
+function computeInWorker(type: WorkerRequestType) {
+  return async (file: Blob): Promise<string> => {
+    if (!worker) {
+      try {
+        worker = new Worker(new URL('./worker', import.meta.url), { type: 'module' })
+        messages = fromEvent<MessageEvent<WorkerResponse>>(worker, 'message')
+      } catch {
+        return compute(type, file)
```

Comment:
I understand that if there isn't a worker and creation fails we want to compute the file on the main thread?

Reply:
Yes I added this because the worker would fail on storybook but it didn't help and I had to alias `@common/utils/crypto/worker` anyway. So I might as well remove this